Web/stockdogweb/src/components/Listing/Listing.js

```diff
@@ -13,55 +13,65 @@ class Listing extends Component {
 
    generateListings = () => {
       var listingElements = [];
+
+      // Dont' do anything if there are no listings
+      if (!this.props.listings) {
```

Comment:
Is the listing null if there is no listing or does it give an empty array? If it's just an empty array I think the forEach won't run because there's nothing in it so you might not need this.

Reply:
Right now both of us don't enforce typechecking for props - [see PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html). This means that one can construct a component without required props. So if the listing prop is left out when using the Listing component, it crashes, thus the check was necessary. If we did use PropTypes, you would be correct in letting forEach do the work.